themes/article/app/routes/$slug[.json].tsx

```diff
@@ -14,7 +14,17 @@ function api404(message = 'No API route found at this URL') {
 
 export const loader: LoaderFunction = async ({ request, params }) => {
   const { slug } = params;
+  // Handle /myst.xref.json as slug
+  if (slug === 'myst.xref') {
+    const xref = await getMystXrefJson();
+    if (!xref) return new Response('MyST Xref not found', { status: 404 });
+    return json(xref);
+  }
   const data = await getPage(request, { slug }).catch(() => null);
   if (!data) return api404('No page found at this URL.');
-  return json(data);
+  return json(data, {
+    headers: {
+      'Access-Control-Allow-Origin': '*',
```

Comment:
This covers the basic case of loading cross-reference tool-tips from external sites. It won't work for github pages...